src/GraphEditor.js

```diff
@@ -363,22 +430,49 @@ export default class GraphEditor extends React.Component {
                     </div>
                 </div>
                 <hr/>
-                <button type="button"
-                    className="btn btn-primary btn-sm"
-                    onClick={this.handleSaveGraph.bind(this)}>Save</button>
-                {this.props.gId &&
-                        <a role="button"
-                            className="btn btn-danger btn-sm float-md-right"
-                            href={"/graph/" + this.props.gId + "/delete/"}>Delete Graph</a>}
-                    </form>
-                </div>;
+                <div className="row">
+                    <div className="ml-3  mr-2">
+                        <button type="button"
+                            className="btn btn-primary"
+                            onClick={this.handleSaveGraph.bind(this)}>Save</button>
+                    </div>
+                    {this.props.gId &&
+                            <div>
+                                <button onClick={this.handleSaveGraph.bind(this)}
+                                    href={"/graph/" + this.props.gId + "/public/"}
```

Comment:
This isn't really going to work - either the onClick or the href will take precedence and prevent the other action. To do a save, then a redirect, you would extend handleSaveGraph to forward to the graph's public view after it's done saving.

I'm guessing we'll talk about this at noon, but I think having "Save" and "Save and View" is awkward. What's wrong with just "Save" and "Preview"? "Save and View" seems like a complicated solution to a really minor problem.

Reply:
The problem with only "Preview" was that if someone made changes to the graph, and then hit preview, it wouldn't preserve the edits they made.  It would show graph at the last save, and navigating back wouldn't show the edits that you wanted to preview.  Also, the meaning of preview is overloaded - the graph is being updated on the edit page, so really there's nothing to preview.  The intention of the user is to see it as a student would, so perhaps "Student View" would be more clear.

I think would be easier for the user if the site saved and redirected to the student view so that they don't need to keep in mind the state of their edits, and it more accurately describes what the student view is.

My vote would be to extend handleSaveGraph even if its more work, and hell - I'd like to get more familiar with how that part of React works.

Reply:
Hmm, yeah I figured. I like "Student View" but that doesn't make it clear that it saves the graph as well. I'm fine with Save and View that was just my initial thought.

I'm looking at handleSaveGraph now - all it does is call saveGraph. What I would do is make a new function called handleViewAndSaveGraph(), copying what's in handleSaveGraph, and then add a redirect there. It may not work correctly off the bat since you might need to use promises to make sure everything happens in the right order - i can help with that.

Reply:
ok cool, I'll take a hack at it

Reply:
cool - and actually call it handleSaveGraphAndView() so it reflects the order of events.